Library/Homebrew/reinstall.rb

```diff
@@ -84,7 +84,7 @@ def self.reinstall_formula(
       keg = install_context.keg
       formula = install_context.formula
       options = install_context.options
-      link_keg = keg.linked?
+      link_keg = keg.linked? if keg&.linked? || nil
```

Comment:
```suggestion
      link_keg = keg&.linked?
```
Isn't this equivalent? Why would `keg` be `nil` here?

Reply:
following the previous flow 
https://github.com/Homebrew/brew/blob/main/Library/Homebrew/reinstall.rb#L26
here the keg value can be unset

Reply:
you were right, both are equivalent and its working as expected